include/aws/common/byte_buf.h

```diff
@@ -128,18 +138,18 @@ int aws_byte_buf_split_on_char(struct aws_byte_buf *input_str, char split_on, st
  * It is the user's responsibility to make sure the input buffer stays in memory long enough to use the results.
  */
 AWS_COMMON_API
-int aws_byte_buf_split_on_char_n(
-    struct aws_byte_buf *input_str,
+int aws_byte_cursor_split_on_char_n(
+    const struct aws_byte_cursor *AWS_RESTRICT input_str,
     char split_on,
-    struct aws_array_list *output,
+    struct aws_array_list *AWS_RESTRICT output,
     size_t n);
 
 /**
  * Copies from to to. If to is too small, AWS_ERROR_DEST_COPY_TOO_SMALL will be
  * returned. dest->len will contain the amount of data actually copied to dest.
```

Comment:
Document that `to == from` is permitted.

Reply:
Will do